x-pack/filebeat/input/s3/input.go

```diff
@@ -372,15 +375,49 @@ func (p *s3Input) handleS3Objects(svc s3iface.ClientAPI, s3Infos []s3Info, errC
 		objectHash := s3ObjectHash(s3Info)
 
 		// read from s3 object
-		reader, err := p.newS3BucketReader(svc, s3Info)
+		resp, err := p.getS3ObjectResponse(svc, s3Info)
 		if err != nil {
 			err = errors.Wrap(err, "newS3BucketReader failed")
+			p.logger.Error(err)
 			s3Context.setError(err)
 			return err
 		}
 
-		if reader == nil {
-			continue
+		if resp == nil {
+			resp.Body.Close()
+			return nil
+		}
+
```

Comment:
I would add `defer resp.Body.Close()` here, and remove the rest of close calls

Reply:
Hmmm this code is inside a for loop for each s3 log https://github.com/elastic/beats/pull/15590/files#diff-ede3f69796ae7b221405174f187a241dR374 (even though s3Infos only should have one set of s3Info in there). That's why I didn't use defer here.